pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientCnx.java

```diff
@@ -399,24 +408,39 @@ protected void handleReachedEndOfTopic(CommandReachedEndOfTopic commandReachedEn
         }
     }
 
-    private boolean addPendingLookupRequests(long requestId, CompletableFuture<LookupDataResult> future) {
-        if (pendingLookupRequestSemaphore.tryAcquire()) {
-            pendingLookupRequests.put(requestId, future);
-            eventLoopGroup.schedule(() -> {
-                if (!future.isDone()) {
-                    future.completeExceptionally(new TimeoutException(
-                            requestId + " lookup request timedout after ms " + operationTimeoutMs));
-                }
-            }, operationTimeoutMs, TimeUnit.MILLISECONDS);
-            return true;
-        }
-        return false;
+    private void addPendingLookupRequests(long requestId, CompletableFuture<LookupDataResult> future) {
```

Comment:
better to leave a comment that the caller of this method needs to be protected under a semaphore